rfcs/text/0352-graph-type-system.md

```diff
@@ -0,0 +1,2659 @@
+
+As the `name` of a Property Type is not guaranteed to be globally unique, the representation of entities is likely going to need to change, especially with regards to changing the keys to be the Property Type URIs:
+
+**An example of an entity in the current system:**
+
+```json
+{
+  "name": "Arthur Philip Dent",
+  "age": 30,
+  "planetOfOrigin": "Earth",
+  "occupation": "Intergalactic Traveler"
+}
+```
+
+**A simplified example of an entity in the proposed system:**
+
+```json
+{
+  "http://example.com/property-types/name": "Arthur Philip Dent",
+  "http://example.com/property-types/age": 30,
+  "http://example.com/property-types/planetOfOrigin": "Earth",
+  "http://example.com/property-types/occupation": "Intergalactic Traveler"
+}
+```
+
+> 💭 Acknowledgement: This is less ergonomic than the current state of things; this drawback among others, is explored with some possible mitigations in the [Drawbacks](#drawbacks), [Rationale and Alternatives](#rationale-and-alternatives), and [Future Possibilities](future-possibilities) sections
+
+This will have an impact on how developers:
+
+- **access** fields in entities
+- format properties when they **create** entities
+- format properties when they **update** entities
+
+### Interfacing with Linked Entities
+
+#### Receiving Links and Linked Entities
+
+Links will likely continue to be returned in a separate collection alongside entities when blocks receive data. The data within the `linkGroups` object might change to incorporate the new type information, likely with the `path` field being replaced with something that uses the link type's URI.
+
+#### Creating Links
+
+The Link Functions will be updated to handle Link Types, removing the `path` fields and instead use the URI of the Link Type.
+
+### Block Schemas
+
+Block schemas will need to be updated to be built using the types outlined above. The specifics of this are explored in-depth in the Reference-Level explanation, however the basic premise is that instead of writing new definitions inline in every block schema, the properties and links should be defined as references to their respective types. If a type does not exist for a given property then it will need to be created, and so on.
+
+### Structure-based Queries
+
+Queries on the structure of data (those which do not directly reference an exact Entity Type), such as `AggregateEntities`, `AggregateEntityType` and any future methods the Block Protocol supports, behave somewhat similarly to expressing a type. Requiring combinations of properties, constraints, etc. will need to refer to the respective Property Types, Link Types, etc.
+
+### Link Constraints
+
+This RFC does not currently mention ways of constraining the destination of a Link Type used in an Entity Type. **This is different from current way of handling things**. The detailed reasoning for removing this, and a potential solution, is outlined in a [follow-up RFC](https://github.com/blockprotocol/blockprotocol/pull/354).
```

Comment:
Going to leave this thread open as it's an important point. Perhaps we go with your prior suggestion and continue to allow pointing to an exact type, and address the problem with the other RFC rather than introducing some sort of implementation dependency.